packages/docusaurus-theme-common/src/utils/generalUtils.ts

```diff
@@ -17,3 +21,70 @@ export function useTitleFormatter(title?: string | undefined): string {
     ? `${title.trim()} ${titleDelimiter} ${siteTitle}`
     : siteTitle;
 }
+interface HasLabel {
+  label: string;
+}
+
+interface HasName {
+  name?: string;
+  imageURL?: string;
+}
+
+type Entry = HasLabel | HasName;
+
+/**
+ * Takes a list of tags or author (as provided by the content plugins),
+ * and groups them by their initials.
+ */
+function listByLetters<T extends Entry, R>(
+  items: readonly T[],
+  getLetter: (item: T) => string,
+  getLabel: (item: T) => string,
+  mapResult: (letter: string, items: T[]) => R,
```

Comment:
Wonder if we really need to map the result

We could just have a result like `LetterEntry = {letter: string | undefined, items: T[]`

And use this directly in the component